src/waitlist.cpp

```diff
@@ -95,36 +134,25 @@ bool WaitingList::clientLogin(const Player* player)
 		return false;
 	}
 
-	slot = priorityWaitList.size();
+	slot = info->priorityWaitList.size();
 	if (player->isPremium()) {
-		priorityWaitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
+		info->priorityWaitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
 	} else {
-		slot += waitList.size();
-		waitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
+		slot += info->waitList.size();
```

Comment:
Not really related, but a list specifically has constant complexity for every single operation we use (`emplace_back` and `erase`), even though it is horribly managed on memory.

A priority queue would do little more than two queues here, just spare a pointer and a size value.